pkg/util/httprequest/httprequest_test.go

```diff
@@ -156,7 +135,7 @@ func TestSchemeHost(t *testing.T) {
 				},
 			},
 			expectedScheme: "http",
-			expectedHost:   "route-namespace.router.default.svc.cluster.local:80",
```

Comment:
based on @miminar's explanation, this was likely captured while our router had the port-stripping bug. Can you set up a pod and access it through the current haproxy router and verify what `X-Forwarded-Host` is set to when accessing http://route, http://route:80, https://route and https://route:443?